src/GridFS/StreamWrapper.php

```diff
@@ -280,8 +292,24 @@ private function getStatTemplate(): array
      */
     private function initProtocol(string $path): void
```

Comment:
If the changes below remain, `initProtocol()` is no longer an accurate name for this method. Something like `initProtocolAndContextOptions()` would be more appropriate.

I'm not a fan of initializing the union of context options for readable and writable streams, but I think that can be revisited once the implementation is actually working (noting the issues with inferring a filename and querying for an `fs.files` document).

Reply:
Function removed.